Cython/Compiler/Scanning.py

```diff
@@ -538,3 +552,29 @@ def exit_async(self):
             self.keywords.discard('async')
             if self.sy in ('async', 'await'):
                 self.sy, self.systring = IDENT, self.context.intern_ustring(self.sy)
+
+@contextmanager
+def tentatively_scan(scanner):
+    errors = hold_errors()
+    try:
+        put_back_on_failure = scanner.put_back_on_failure
+        scanner.put_back_on_failure = []
+        initial_state = (scanner.sy, scanner.systring, scanner.position())
+        try:
+            yield errors
+        except CompileError as e:
+            pass
+        finally:
+            if errors:
+                for put_back in reversed(scanner.put_back_on_failure[:-1]):
```

Comment:
Not sure which is better here.
```suggestion
                for put_back in scanner.put_back_on_failure[-2::-1]:
```

Reply:
I find this slightly harder to read I think